netmiko/garderos/garderos_ssh.py

```diff
@@ -0,0 +1,488 @@
+            output_summary += output_current
+            if "exited" in output_current:
+                # SSH session terminated
+                # Clear the receive channel before raising exception // Also retrieves error message
+                output_current = self.read_channel()
+                raise ConnectionError(
+                    "Error connecting to host {}. Session terminated unexpectedly. \n{}".format(
+                        host, output_current.strip()
+                    )
+                )
+            elif not password_sent and "user" in output_current.lower():
+                # Send username when prompted
+                self.write_channel(username + "\n")
+            elif not password_sent and "password" in output_current.lower():
+                # Send password when prompted for the first time
+                self.write_channel(password + "\n")
+                password_sent = True
+            elif password_sent and "password" in output_current.lower():
+                # If password has already been sent but a second password prompt appears,
+                # this means the password was incorrect
+                # Cancel nested SSH connection: CTRL+C followed by ENTER
+                self.write_channel(b"\x03".decode("utf-8"))
+                sleep(1)
+                self.write_channel("\n")
+                sleep(2)
+                # Clear the receive channel before raising exception
+                output_current = self.read_channel()
+                raise ValueError(
+                    "Authentication error while connecting to host {}.".format(host)
+                )
+            elif (
+                password_sent
+                and not login_completed
+                and not (expect_prompt in output_current)
+            ):
+                # After password has been sent, try to send ENTER to device to get the device prompt
+                self.write_channel("\n")
+            elif (
+                password_sent
+                and not login_completed
+                and expect_prompt in output_current
+            ):
+                # Device prompt successfully received
+                login_completed = True
+            # Increase counter in any case
+            counter += 1
+        # On exit of loop check whether password has been sent
+        if not password_sent or not login_completed:
+            # Cancel nested SSH connection: CTRL+C followed by ENTER
+            self.write_channel(b"\x03".decode("utf-8"))
+            sleep(1)
+            self.write_channel("\n")
+            sleep(2)
+            # Clear the receive channel before raising exception
+            output_current = self.read_channel()
+            raise TimeoutError(
+                "Timeout error while connecting to host {}. Transcript of session: \n{}".format(
+                    host, output_summary
+                )
+            )
+        return output_current.strip()
+
+    def ssh_send_command(
```

Comment:
Purpose? Probably should be outside of Netmiko driver (i.e. in end-user code).

Reply:
See above.

Reply:
Removed in updated PR